homeassistant/components/huum/coordinator.py

```diff
@@ -0,0 +1,58 @@
+"""DataUpdateCoordinator for Huum."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from huum.exceptions import Forbidden, NotAuthenticated
+from huum.huum import Huum
+from huum.schemas import HuumStatusResponse
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+type HuumConfigEntry = ConfigEntry[HuumDataUpdateCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+UPDATE_INTERVAL = timedelta(seconds=30)
+
+
+class HuumDataUpdateCoordinator(DataUpdateCoordinator[HuumStatusResponse]):
+    """Class to manage fetching data from the API."""
+
```

Comment:
```suggestion
    config_entry: HuumConfigEntry
```
that's why mypy is complaining